### PR TITLE
Add SonicDBConfig.getInstanceList() API

### PR DESCRIPTION
**Pull request by liuh-80**
#### Why I did it
SonicDBConfig.getInstanceList() API is missing in swsscommon.
The py-swsssdk library are going to be deprercated, to change all script to use swsscommon, we need add this missing API.

#### How I did it
Add SonicDBConfig.getInstanceList() API.

#### How to verify it
Pass all existed test case
Add new test case to cover new API.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

#### Description for the changelog
Add SonicDBConfig.getInstanceList() API.

### CODE REVIEW

File: pyext/swsscommon.i [hunk 1]
```diff
@@ -83,15 +84,6 @@
     }
 }
 
-%template(FieldValuePair) std::pair<std::string, std::string>;
```
Comment (liuh-80) [inline pyext/swsscommon.i:86]: All these removed codes are dupe with codes between line 52 -- 58.

File: common/dbconnector.cpp [hunk 2]
```diff
@@ -391,6 +391,27 @@ std::vector<std::string> SonicDBConfig::getDbList(const std::string &netns)
...
 
+map<string, RedisInstInfo> SonicDBConfig::getInstanceList(const std::string &netns)
+{
+    if (!m_init)
+    {
+        initialize();
+    }
+    validateNamespace(netns);
+
+    map<string, RedisInstInfo> result;
```
Comment (liuh-80) [inline common/dbconnector.cpp:402]: Not return unordered_map here, because map already mapping to python dictionary.

File: common/dbconnector.h [hunk 3]
```diff
@@ -81,23 +82,24 @@ class SonicDBConfig
     static std::vector<std::string> getDbList(const std::string &netns = EMPTY_NAMESPACE);
     static bool isInit() { return m_init; };
     static bool isGlobalInit() { return m_global_init; };
+    static std::map<std::string, swss::RedisInstInfo> getInstanceList(const std::string &netns = EMPTY_NAMESPACE);
 
 private:
     static std::recursive_mutex m_db_info_mutex;
     // { namespace { instName, { unix_socket_path, hostname, port } } }
-    static std::unordered_map<std::string, std::unordered_map<std::string, RedisInstInfo>> m_inst_info;
+    static std::unordered_map<std::string, std::unordered_map<std::string, swss::RedisInstInfo>> m_inst_info;
```
Comment (qiluo-msft) [inline common/dbconnector.h:90]: >swss:: [](http://example.com/codeflow?start=75&length=6)

Prefix is not needed.

File: common/dbconnector.cpp [hunk 4]
```diff
@@ -391,6 +391,27 @@ std::vector<std::string> SonicDBConfig::getDbList(const std::string &netns)
...
+    auto iterator = m_inst_info.find(netns);
```
Comment (qiluo-msft) [inline common/dbconnector.cpp:403]: >m_inst_info [](http://example.com/codeflow?start=20&length=11)

Is it better to change m_inst_info type to unordered_map?

File: common/dbconnector.h (hunk 3 above)
Comment (liuh-80) [reply common/dbconnector.h]: Fixed.

File: common/dbconnector.cpp (hunk 4 above)
Comment (liuh-80) [reply common/dbconnector.cpp:408]: Fixed, change to use map.

I try to use unordered_map and include the std_unordered_map.i in swsscommon.i
However SWIG build report following error:

swsscommon_wrap.cpp:4065:76: error: 'type_name' is not a member of 'swig::traits<std::unordered_map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> > >'
     return traits<typename noconst_traits<Type >::noconst_type >::type_name();
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~

**Comment by qiluo-msft** [comment]
> class RedisInstInfo

Why change this class?

---
Refers to: common/dbconnector.h:22 in e9500b2. [](commit_id = e9500b2a23a50af83fa138314fa7871bf88ed2af, deletion_comment = True)

**Comment by liuh-80** [comment]
> > class RedisInstInfo
> 
> Why change this class?
> 
> Refers to: common/dbconnector.h:22 in [e9500b2](https://github.com/Azure/sonic-swss-common/commit/e9500b2a23a50af83fa138314fa7871bf88ed2af). [](commit_id = [e9500b2](https://github.com/Azure/sonic-swss-common/commit/e9500b2a23a50af83fa138314fa7871bf88ed2af), deletion_comment = True)

I will change RedisInstInfo back.

Seems I misunderstand 'Is it better to change m_inst_info type to unordered_map?', the reason change this is because the py-swsscommon version of this API return redis info as a hashmap.

File: common/dbconnector.h [hunk 5]
```diff
@@ -13,6 +14,10 @@
 #include "redisreply.h"
 #define EMPTY_NAMESPACE std::string()
 
+#define REDIS_UNIX_SOCKET_PATH "UnixSocketPath"
+#define REDIS_HOSTNAME "Hostname"
+#define REDIS_PORT "Port"
```
Comment (qiluo-msft) [inline common/dbconnector.h:19]: Why adding these constants?

Comment (liuh-80) [reply common/dbconnector.h]: Fixed.
Forgot to revert this change, this change added when change RedisInstInfo to map.

**Comment by liuh-80** [comment]
This need cherry-pick to 202205 to fix this issue: https://github.com/sonic-net/sonic-buildimage/issues/12050

**Pull request merged**